source/production/F0.Minesweeper.Components/Difficulty.razor.cs

```diff
@@ -0,0 +1,45 @@
+using System.Threading.Tasks;
+using F0.Minesweeper.Components.Abstractions.Enums;
+using Microsoft.AspNetCore.Components;
+
+namespace F0.Minesweeper.Components
+{
+	public partial class Difficulty
+	{
+		private DifficultyLevel selected;
+
+		[Parameter]
+		public EventCallback<DifficultyLevel> OnChange { get; set; }
+
+		public DifficultyLevel Selected
+		{
+			get => selected;
+			set
+			{
+				if (selected != value)
```

Comment:
The way the setter is written the `selected` field will also be set in case the value didn't change which is kind of unnecessary. To prevent such things I personally prefer *gated* checks. So basically checking here:
```
if (selected == value)
{
    return;
}

// do rest of the code
```

Reply:
sure thing! will do